taichi/codegen/codegen_llvm.cpp

```diff
@@ -1552,6 +1552,14 @@ void CodeGenLLVM::create_offload_struct_for(OffloadedStmt *stmt, bool spmd) {
     auto upper_bound = get_arg(4);
 
     parent_coordinates = element.get_ptr("pcoord");
+    block_corner_coordinates =
+        create_entry_block_alloca(physical_coordinate_ty);
+
+    auto refine =
+        get_runtime_function(leaf_block->refine_coordinates_func_name());
+
+    create_call(refine, {parent_coordinates, block_corner_coordinates,
+                         tlctx->get_constant(0)});
```

Comment:
Why is this `0`?

Reply:
The block corner has the same coordinates as the leaf element of the first (0th) loop_index . And it is the same for any loop_index. The refine function here basically just left shifts the parent coordinates by num_bits.